### PR TITLE
Support bring-your-own-openai resource

### PR DESCRIPTION
**Pull request by jamesrcounts**
# Allow customers to bring an existing OpenAI account

## The issue or feature being addressed

Support the following scenarios:
- Customers might not have access to OpenAI in the subscription to which they plan to deploy FoundationaLLM.  
- Customers may have an existing OpenAI instance and do not wish to split deployment quotas between the existing instance and a dedicated FoundationaLLM instance.

## Details on the issue fix or feature implementation

This PR updates the quickstart ACA template to accept an existing Azure OpenAI instance in the same or different subscription for use with FoundationaLLM.

## Confirm the following

- [x]  I started this PR by branching from the head of the default branch
- [x]  I have targeted the PR to merge into the default branch
- [ ]  I have included unit tests for the issue/feature
- [x]  I have included inline docs for my changes, where applicable
- [x]  I have successfully run a local build
- [x]  I have provided the required update scripts, where applicable
- [x]  I have updated relevant docs, where applicable

### CODE REVIEW

**Review by hatboyzero — APPROVED** [review]

File: deploy/starter/README.md
Comment (ciprianjichici) [inline deploy/starter/README.md:1]: Are we still requiring Docker Desktop?

Comment (jamesrcounts) [reply deploy/starter/README.md:1]: It is only needed for those who want to run and debug locally, the same as Visual Studio and dotnet sdk.

Comment (jamesrcounts) [reply deploy/starter/README.md:1]: added

[the remark this answers is not in the document]
Comment (jamesrcounts) [reply deploy/starter/README.md:1]: added

**Review by ciprianjichici — APPROVED** [review]

**Pull request merged**